### PR TITLE
Correctly fallback on service:plan add-on resolution

### PR DESCRIPTION
**Pull request by bjeanes**
`heroku addons:open -a APP ADDON_SERVICE` was failing because it would first look up `ADDON_SERVICE` as an add-on name. heroku/api#4725 meant that this endpoint returned something other than a 404 for a string like `papertrail`. This accounts for that. Fortunately heroku/api#4725 actually means this entire resolver can be simplified drastically as most of this resolution logic now lives in the API. I'll handle that in a separate PR, though, as it'll take more time.

cc @heroku/add-ons @dickeyxxx

### CODE REVIEW

File: lib/heroku/helpers/addons/resolve.rb [hunk 1]
```diff
@@ -85,16 +82,23 @@ def resolve_addon(identifier, &filter)
...
             rescue Heroku::API::Errors::Forbidden
               # treat permission error as no match because there might exist a
               # resource on someone else's app that has a name which
               # corresponds to a service name that we wish to check below (e.g.
               # "memcachier")
+            rescue Heroku::API::Errors::RequestFailed => e
+              error_id = (json_decode(e.response.body.to_s) || {})["id"]
+              if error_id == "multiple_matches"
+                # ignore; name-based matches are never ambiguous, so this
+                # identifier is probably a service/plan
```
Comment (mnoble) [inline lib/heroku/helpers/addons/resolve.rb]: For the `ADDON_SERVICE` scenario, isn't the above only true when the request is scoped to an app? If they did `heroku addons:open papertrail` in a non-app directory, don't we still want the "Ambiguous identifier" error?

Comment (bjeanes) [inline lib/heroku/helpers/addons/resolve.rb]: Yeah it's tricky. The next piece will prompt for `app` which will error and ask for `--app`, so I think it always ends up in the app-scoped case when using those identifiers. We *could* have it be an ambiguous match for all the `papertrail`s across all app, but the way to handle that will be to ask for an app anyway, so it's really just moving the error. In any case, #1706 should make all of this more sane. What do you think?

**Comment by bjeanes** [comment]
@dickeyxxx bump! getting a few tickets about this.

File: lib/heroku/helpers/addons/resolve.rb (hunk 1 above)
Comment (mattreduce) [inline lib/heroku/helpers/addons/resolve.rb]: A-ha, thanks for explaining. That should do for now (and #1706 will help when it lands) :+1:

**Comment by mattreduce** [comment]
Looks good to me, thanks @bjeanes!

**Pull request merged**